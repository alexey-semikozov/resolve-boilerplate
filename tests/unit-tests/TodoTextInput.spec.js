import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import renderer from 'react-test-renderer';
import TodoTextInput from '../../components/TodoTextInput';

const setup = propOverrides => {
  const props = Object.assign(
    {
      onSave: jest.fn(),
      text: 'Use Redux',
      placeholder: 'What needs to be done?',
      editing: false,
      newTodo: false
    },
    propOverrides
  );

  const shallowrenderer = new ReactShallowRenderer();
  shallowrenderer.render(<TodoTextInput {...props} />);
  let output = shallowrenderer.getRenderOutput();

  const component = renderer.create(<TodoTextInput {...props} />);
  const tree = component.toJSON();

  return {
    props,
    output,
    tree
  };
};

describe('components', () => {
  describe('TodoTextInput', () => {
    it('should render correctly', () => {
      const { tree } = setup();
      expect(tree).toMatchSnapshot();
    });

    it('should render correctly when editing=true', () => {
      const { output } = setup({ editing: true });
      expect(output.props.className).toEqual('edit');
    });

    it('should render correctly when newTodo=true', () => {
      const { output } = setup({ newTodo: true });
      expect(output.props.className).toEqual('new-todo');
    });
  });
});
