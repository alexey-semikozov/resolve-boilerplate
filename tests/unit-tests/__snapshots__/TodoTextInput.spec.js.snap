// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components TodoTextInput should render correctly 1`] = `
<input
  autoFocus="true"
  className=""
  onBlur={[Function]}
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="What needs to be done?"
  type="text"
  value="Use Redux"
/>
`;
